src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,52 @@
 'use strict';
 
+const BASE_HEALTH = 100;
+const CARNIVORE_BITE_FORCE = 50;
+const DEATH_HEALTH = 0;
+
 class Animal {
-  // write your code here
+  static alive = [];
+  health = BASE_HEALTH;
+
+  constructor(name) {
+    this.name = name;
+  }
+
+  static removeFromAlive(creature) {
+    this.alive = this.alive.filter((animal) => animal !== creature);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= CARNIVORE_BITE_FORCE;
+
+      if (herbivore.health <= DEATH_HEALTH) {
+        Animal.removeFromAlive(herbivore);
+      }
```

Comment:
[CODE PERFORMANCE] It is preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method. Here, you are using a separate method to remove a dead animal which is acceptable, but could be less performant if called frequently as it updates the `Animal.alive` array multiple times. Consider filtering the `Animal.alive` array in one go after all bite operations are performed to optimize performance.